cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -197,10 +197,16 @@ public synchronized void delete(final Id.Namespace namespaceId) throws Exception
       // Another reason for not deleting the default namespace is that we do not want to call a delete on the default
       // namespace in the storage provider (Hive, HBase, etc), since we re-use their default namespace.
       if (!Id.Namespace.DEFAULT.equals(namespaceId)) {
-        // Finally delete namespace from MDS
-        nsStore.delete(namespaceId);
-        // Delete namespace in storage providers
-        dsFramework.deleteNamespace(namespaceId);
+        NamespaceMeta namespaceMeta = nsStore.get(namespaceId);
+        try {
+          dsFramework.deleteNamespace(namespaceId);
+        } catch (DatasetManagementException ex) {
+          LOG.error("Error while trying to delete the namespace {} (whose config is {}) in dataset framework",
+                    namespaceId.getId(), namespaceMeta.getConfig());
+        } finally {
+          // Finally delete namespace from MDS
+          nsStore.delete(namespaceId);
```

Comment:
I see that it will always execute this statement since its in a finally, but the difference from before is that the API response will not contain the error now?